src/v2/Artsy/Analytics/trackingMiddleware.ts

```diff
@@ -109,6 +115,19 @@ export function trackingMiddleware(options: TrackingMiddlewareOptions = {}) {
               }
             })
           }
+
+          // AB Test
+          abTestRouteMap.forEach(({ abTest, routes }) => {
+            routes.some(route => {
+              const matcher = match(route, { decode: decodeURIComponent })
+              const foundMatch = !!matcher(pathname)
+
+              if (foundMatch) {
+                trackExperimentViewed(abTest)
+                return true
+              }
+            })
+          })
```

Comment:
If not for this, client-side navigation would not fire the experiment-viewed event?

Reply:
Correct, this is the snippet that actually fires the event. An alternative might be to put something custom in on the artist page app itself, but it kinda felt nice to make something a bit more generic for this, especially since it piggy backs into the tracking middleware (which actually feels appropriate too).